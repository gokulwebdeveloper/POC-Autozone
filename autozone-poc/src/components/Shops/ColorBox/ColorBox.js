import React, { Fragment } from 'react';
import PropTypes from 'prop-types';


ColorBox.propTypes = {
    colors: PropTypes.array,
    colors: PropTypes.any.isRequired,
}

// Specifies the default values for props:
ColorBox.defaultProps = {
    colors: ['color1', 'color2']
};

export default function ColorBox(props) {

    function ColorBoxItem(props) {

        return (
            <li><a href="#" className={props.value}></a></li>
        );
    }

    if (props != undefined && props.colors != undefined) {
        return (
            <div className="widget color mb-50">
                <p className="widget-title2 mb-30">Color</p>
                <div className="widget-desc">
                    <ul className="d-flex" >
                        {props.colors.map((item) => {
                            return <ColorBoxItem value={item} key={item.toString()} />
                        }
                        )}
                    </ul>
                </div>
            </div>
        );
    } else {
        return <p>no data</p>
    }

}