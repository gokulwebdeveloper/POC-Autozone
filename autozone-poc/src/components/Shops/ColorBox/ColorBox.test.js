import '@testing-library/jest-dom/extend-expect';
import renderer from "react-test-renderer";
import React, { Component } from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { render } from '@testing-library/react';
import ColorBox from './ColorBox';

configure({ adapter: new Adapter() });


describe('Brand testing', () => {
    var colors = [
		"color1",
		"color2",
		"color3",
		"color4",
		"color5",
		"color6",
		"color7",
		"color8",
		"color9",
		"color10"
	];


    let wrapper;
    wrapper = shallow(<ColorBox colors={colors}/>);

    it("SnapShot Match for Brand component", () => {
        const page = renderer.create(<ColorBox colors={colors}/>).toJSON();
        expect(page).toMatchSnapshot();
      });

   

  })