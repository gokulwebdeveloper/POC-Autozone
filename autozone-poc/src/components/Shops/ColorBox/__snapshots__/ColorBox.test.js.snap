// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand testing SnapShot Match for Brand component 1`] = `
<div
  className="widget color mb-50"
>
  <p
    className="widget-title2 mb-30"
  >
    Color
  </p>
  <div
    className="widget-desc"
  >
    <ul
      className="d-flex"
    >
      <li>
        <a
          className="color1"
          href="#"
        />
      </li>
      <li>
        <a
          className="color2"
          href="#"
        />
      </li>
      <li>
        <a
          className="color3"
          href="#"
        />
      </li>
      <li>
        <a
          className="color4"
          href="#"
        />
      </li>
      <li>
        <a
          className="color5"
          href="#"
        />
      </li>
      <li>
        <a
          className="color6"
          href="#"
        />
      </li>
      <li>
        <a
          className="color7"
          href="#"
        />
      </li>
      <li>
        <a
          className="color8"
          href="#"
        />
      </li>
      <li>
        <a
          className="color9"
          href="#"
        />
      </li>
      <li>
        <a
          className="color10"
          href="#"
        />
      </li>
    </ul>
  </div>
</div>
`;
